Adding area toggle to top of docs nav

phazonoverload opened the pull request:
There are two main areas of our docs - the developer reference and the user guide. @benhaynes has commented that it's hard to understand that the current header navigation links take you between these areas (sidebars). 

This PR adds tabs to the top of the sidebar, removes them from the top nav, and also adds Twitter/Discord/Directus TV to the top nav. 

![2024-01-24T0947 23](https://github.com/directus/directus/assets/1461554/66395e4c-36eb-4a84-9146-d6db122101b8)

phazonoverload committed:
Added tabs

phazonoverload committed:
Remove area links from nav

phazonoverload committed:
Added social nav links

phazonoverload committed:
Ensured toggle is respected when path changes

phazonoverload committed:
Updated dev toggle link

phazonoverload committed:
Removed white backgroudn to respect dark mode

phazonoverload committed:
CI gods must be pleased

paescuj committed:
Simplify path determination

benhaynes commented:
@phazonoverload — the purple on gray doesn't work for me in that button, and I often times get confused when there are two buttons with a single active state (if one is white and the other black, which is active??). I think we should look into a more obvious active toggle state, such as an outline vs fill... or even using an icon (`check`). Mock below:

![CleanShot 2024-01-24 at 11 15 46](https://github.com/directus/directus/assets/522079/4c365577-0683-409b-87de-05517a0e6e03)

phazonoverload commented:
We can explore that but we're working with very little horizontal space. I borrowed colors/styling from the tab toggle on the docs home. I'll try getting closer to your mock and update here. 
![2024-01-24T1620 21](https://github.com/directus/directus/assets/1461554/5afa2079-8585-4e9e-8f9e-d9d12675097c)

phazonoverload committed:
Toggle design as per Ben's feedback

phazonoverload committed:
Tabs component styling

phazonoverload committed:
Updated tabs content on docs home

phazonoverload committed:
HEader nav links

phazonoverload committed:
Dev sidebar link to home

phazonoverload commented:
This works great @benhaynes 

![2024-01-25T0805 23](https://github.com/directus/directus/assets/1461554/91c41f98-4143-4e93-966b-38aa0187916b)

While I was here I also updated the tab styling to match on the docs landing page

![2024-01-25T0808 29](https://github.com/directus/directus/assets/1461554/9cb2701a-f516-4036-87c1-3e85a33823cf)

And updated the header link text based on suggestions from @paescuj 

🖖🏻

paescuj committed:
Adjust inactive & hover color of tab buttons

paescuj committed:
Nest CSS

phazonoverload commented on docs/.vitepress/theme/Layout.vue:
```diff
@@ -40,7 +51,35 @@ const isPackagePage = RegExp('^/packages/.+$').test(path.value);
+			padding: 0.25rem;
+			font-weight: bold;
+			border-radius: 10rem;
+			border: 1px solid var(--vp-input-border-color);
+			color: var(--vp-input-border-color);
+
```
Better!

phazonoverload committed:
Update docs/.vitepress/theme/Layout.vue

Co-authored-by: Pascal Jufer <pascal-jufer@bluewin.ch>

benhaynes commented:
This works, thank you @phazonoverload! Now that the website is done, we can plan on going through the Docs website all at once (once we've done a formal assessment of its structure/content), but this is a good change for now. 🙌

phazonoverload commented:
Yup let us scope out the non design work first. It could be a rebuild. It might just be smaller work. Discovery is happening in the next couple of months.
